### PR TITLE
Add new board NUCLEO_F722ZE

### PR DESCRIPTION
**Pull request by badlyby**
Add definitions that support MCU STM32F722xx.

### CODE REVIEW

File: ports/stm32/boards/NUCLEO_F722ZE/pins.csv [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+UART2_TX,PD5
+UART2_RX,PD6
+UART2_RTS,PD4
+UART2_CTS,PD3
+UART6_TX,PG14
+UART6_RX,PG9
+SPI_B_NSS,PA4
+SPI_B_SCK,PB3
+SPI_B_MOSI,PB5
+ETH_MDC,PC1
```
Comment (dpgeorge) [inline ports/stm32/boards/NUCLEO_F722ZE/pins.csv]: This MCU doesn't have ETH so these ETH lines can be removed.

**Comment by dpgeorge** [comment]
Thanks for updating it.

What kind of testing did you do, did you test the filesystem?

File: ports/stm32/boards/NUCLEO_F722ZE/pins.csv (hunk 1 above)
Comment (badlyby) [reply ports/stm32/boards/NUCLEO_F722ZE/pins.csv]: This MCU doesn't have ETH PHY，but the board has ETH element.
I'm not sure if I want to keep it.
Now I'll remove it.

**Review by badlyby — COMMENTED** [review]
> Thanks for updating it.
> 
> What kind of testing did you do, did you test the filesystem?

I tested the filesystem.
Other external interfaces are the same as NUCLEO_144, so I haven't tested them all.

**Comment by dpgeorge** [comment]
Ok, merged in 0da2f6f23a8316f6810c47b2d8410bf11029f2a1, 09267bb147c06ef5350b91034342924a9d9efa05 and 3b258ef213d34b402c170fbb63dafbab53ea91b2 (with some minor changes).

**Pull request closed without merging**